packages/app/src/api.ts

```diff
@@ -171,6 +180,126 @@ export function initializeApp(
   return newApp;
 }
 
+/**
+ * Creates and initializes a {@link @firebase/app#FirebaseServerApp} instance.
+ *
+ * The FirebaseServerApp is similar to FirebaseApp, but is intended for execution in
+ * server side rendering environments only. Initialization will fail if invoked from a
+ * browser environment.
+ *
+ * See
+ * {@link
+ *   https://firebase.google.com/docs/web/setup#add_firebase_to_your_app
+ *   | Add Firebase to your app} and
+ * {@link
+ *   https://firebase.google.com/docs/web/setup#multiple-projects
+ *   | Initialize multiple projects} for detailed documentation.
+ *
+ * @example
+ * ```javascript
+ *
+ * // Initialize a FirebaseServerApp.
+ * // Retrieve your own options values by adding a web app on
+ * // https://console.firebase.google.com
+ * initializeServerApp({
+ *     apiKey: "AIza....",                             // Auth / General Use
+ *     authDomain: "YOUR_APP.firebaseapp.com",         // Auth with popup/redirect
+ *     databaseURL: "https://YOUR_APP.firebaseio.com", // Realtime Database
+ *     storageBucket: "YOUR_APP.appspot.com",          // Storage
+ *     messagingSenderId: "123456789"                  // Cloud Messaging
+ *   },
+ *   {
+ *    authIdToken: "Your Auth ID Token"
+ *   });
+ * ```
+ *
+ * @param options - Firebase.AppOptions to configure the app's services, or a
+ *   a FirebaseApp instance which contains the AppOptions within.
+ * @param config - FirebaseServerApp configuration.
+ *
+ * @returns The initialized FirebaseServerApp.
+ *
+ * @public
+ */
+export function initializeServerApp(
+  options: FirebaseOptions | FirebaseApp,
+  config: FirebaseServerAppSettings
+): FirebaseServerApp;
+
+export function initializeServerApp(
+  _options: FirebaseOptions | FirebaseApp,
+  _serverAppConfig: FirebaseServerAppSettings
+): FirebaseServerApp {
+  if (isBrowser()) {
```

Comment:
I think we should include ReactNative, MobileCordova, etc. Actually, the only supported environment should be Node?

Reply:
One of the challenge's @mbleigh gave us was to not be Node specific. I could see developers using "FirebaseServerApps" anywhere that auth signIn is not permitted, e.g, Bun, Deno, edge compute, Chrome Extensions, etc. While NodeJS will be our only "supported" runtime, we don't want to explicitly disallow non-Node environments other than the CSR.